### DISCUSSION TITLE
Switch global themes

### DISCUSSION THREAD

**Discussion by ignacio-gc** [vercel/next.js#15826] [Help]
I want to add a button to switch between dark and light themes, but can't understand how to conditionally apply a global style. I've tried with `<style jsx global>` in `index.tsx`, also in `Layout`  component (from the tutorial) with no luck. Also I couldn't find a way to do it in `_app.ts`.

Thanks for the help.

**Comment by Oikio** [comment]
For styled-components you can use something like this in _app.tsx:
```
interface Props extends AppProps {
  err: Error
}

const Dark = createGlobalStyle`body {background: #000 !important;}`
const Light = createGlobalStyle`body {background: red !important;}`

const App: NextPage<Props> = () => {
  useSentryRouteInfo()
  const [isDark, setIsDark] = useState(false)

  return (
    <>
      {isDark ? <Dark /> : <Light />}
      <button onClick={() => setIsDark(!isDark)}>Switch</button>
    </>
  )
}

export default App
```
But it doesn't matter, it can be any css-in-js solution or even just a dangerously inserted string in <style> tag, it's just what I had opened to test it.
Condition didn't work inside Next's Head tag though, no idea why - I don't know how it's implemented.

**Reply by ignacio-gc** [reply]
Thanks for your answer. Is there a solution without styled components?

**Reply by Oikio** [reply]
As I said before - you can use whatever you want there, as long as it will result in component - you can use it
Also you can use string if it suits you (it all depends on the project). For example, straight and simple:
```
interface Props extends AppProps {
  err: Error
}

const dark = `body {background: #000 !important;}`
const light = `body {background: red !important;}`

const App: NextPage<Props> = () => {
  useSentryRouteInfo()
  const [isDark, setIsDark] = useState(false)

  return (
    <>
      <style dangerouslySetInnerHTML={{ __html: isDark ? dark : light }} />
      <button onClick={() => setIsDark(!isDark)}>Switch</button>
    </>
  )
}

export default App
```

**Comment by ignacio-gc** [comment]
Sorry I'm new to React and Next. Why importing the .css file in `_app.js` and then add `className` to `<Component />` doesn't work? 
I see that something is wrong, but I'm trying to understand why can't be done adding or removing a class attribute.

This is adapted from the tutorial: 
```
import '../styles/global.css'
import { AppProps } from 'next/app'
import '../styles/themes.css'

export default function App({ Component, pageProps }: AppProps) {
    return <Component className="dark" {...pageProps} />
}
```

**Reply by Oikio** [reply]
See if your Component inside passes this className to an actual element, like div\span and etc.
